src/Betfair/Adapter/ArrayRpcAdapter.php

```diff
@@ -14,6 +14,21 @@ class ArrayRpcAdapter implements AdapterInterface
     public function adaptResponse($response)
     {
         $data = json_decode($response, true);
-        return $data['result'];
+        
+        // check if $data contains result from Betfair API
+        
+        if(isset($data['result'])){
+            
+        	return $data['result'];
+        	
+        }
+        
+        // check if $data contains error from Betfair API
+        
+        if(isset($data['error'])){
```

Comment:
Ok! Is it possibile, but it's a bit difficult.
Have a look this class: https://github.com/danieledangeli/betfair-php/blob/master/src/Betfair/Credential/Credential.php

If you build an object "Credential" with a sessionToken it will be used foreach call (without trying to login). 

If think that the session management should be implemented outside the library (e.g store the sesisonToken in some caching layer). But the library should expose a way to build a Credential Object from a token in easy way.

I'll go through that! Thanks!